cmd/kaniko-gcr/main.go

```diff
@@ -155,12 +176,19 @@ func run(c *cli.Context) error {
 		}
 	}
 
+	enableAutoTag := c.Bool("tags.auto")
+	enableExpandTag := c.Bool("tags.expand")
+	tags, shouldSkipBuild, err := tagger.MaybeAutoTag(c.StringSlice("tags"), c.String("commit.ref"), c.String("tags.auto_suffix"), c.String("repo.branch"), enableAutoTag, enableExpandTag)
```

Comment:
If auto-tag is not enabled, then tags will be empty list which is not correct. Also, L179-L181 should be moved to kaniko.go file since it is same for all of dockerhub, gcr & ecr.

Reply:
Limited the breaking changes to users that enable the current auto-tag flag. Behaviors comparison before/after the PR:
* --auto-tag=false: No changes.
* --auto-tag=false,--expand-tag=true,tags=1.0.0:
  * Old behavior: Should not happen. --expand-tag didn't exist.
  * New Behavior: Build with [1,1.0,1.0.0] tags.
* --auto-tag=true
  * Old behavior: Build with the "latest" tag.
  * New behavior: Build with auto detected tags. Abort if auto detection failed.
* --auto-tag=true,tags=latest: same as "--auto-tag=true".
* --auto-tag=true,tags=1.0.0:
  * Old behavior: Build with [1,1.0,1.0.0] tags.
  * New behavior: Abort the build with an error message.
* --auto-tag=true,--expand-tag=true,tags=1.0.0: Abort the build with an error message.

Moved the logic of L179-L181 to Plugin::Exec() in "kaniko.go".